Explorer/Assets/Scripts/ECS/StreamableLoading/Common/AssetPromise.cs

```diff
@@ -80,7 +80,7 @@ public bool TryGetResult(World world, out StreamableLoadingResult<TAsset> result
         /// </summary>
         public bool TryConsume(World world, out StreamableLoadingResult<TAsset> result)
         {
-            Assert.IsFalse(Result.HasValue, ASSERTION_MESSAGE);
+            //Assert.IsFalse(Result.HasValue, ASSERTION_MESSAGE);
```

Comment:
Im concerned about this one. By design it can only be consumed once. It means that we have multiple systems trying to consume the same promise, but only one, the first to consume it, will get a valid result.
Also this change might have a big impact. I would like to re-check with @mikhail-dcl

Reply:
I'll remove this from this PR, but its also a very bad design that a single failure stops the correct loading of everything instead of properly logging the issue

Reply:
Also, why does the design contain a "development only" line of code?

Reply:
Perhaps we should allow the promise to be consumed as many times as you want, but keep the result so its consistent, similar on how `TaskCompletionSource` works, but will require a major rework on the systems

Reply:
That could work, but we also have to analyze which system is consuming the promise more than once which is also a symptom of something going on